source/braille.py

```diff
@@ -1000,8 +987,27 @@ def update(self):
 		self.selectionStart = self.selectionEnd = None
 		self._isFormatFieldAtStart = True
 		self._skipFieldsNotAtStartOfNode = False
-
 		self._endsWithField = False
+		# Get the selection.
+		sel = self._getSelection()
+		if not sel.isCollapsed:
+			self._readingInfo = readingInfo = sel.copy()
+			if self.obj.isSelectionAnchoredAtStart:
+				# The end of the range is exclusive, so make it inclusive first.
+				readingInfo.move(textInfos.UNIT_CHARACTER, -1, "end")
+			# Collapse the selection to the point that is moving.
+			readingInfo.collapse(end=self.obj.isSelectionAnchoredAtStart)
+			# Get the reading unit at the end of the selection.
+			readingInfo.expand(unit)

```

Comment:
As noted in the comment that was previously here:

> ```
> 		# HACK: Some TextInfos only support UNIT_LINE properly if they are based on POSITION_CARET,
> 		# so use the original cursor TextInfo for line and copy for cursor.
> ```

I think this is going to break for certain TextInfos. Affected implementations will only let you get the line if you're working with the caret or selection range. Once you copy the range, that ability is lost. This used to be true for Word, though I think we've since rectified that. However, I'm pretty certain this is still true for MSHTML. Long story short: let `readingInfo` be the TextInfo you create from POSITION_SELECTION and then copy it for `sel`.